### PR TITLE
Add psalm checker

### PR DESCRIPTION
**Pull request by Sasanidas**
Add [Psalm](https://psalm.dev/) syntax check capabilities support for flycheck

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -10077,6 +10078,31 @@ See URL `https://github.com/uber/prototool'."
...
+(flycheck-def-option-var flycheck-psalm-threads nil psalm
+  "If greater than one, Psalm will run analysis on multiple
+threads, speeding things up."
+  :type '(integer :tag "Number of threads in use")
+  :safe #'integerp)
+
+(flycheck-define-checker psalm
+  "Flycheck for Psalm, the free software static tool for PHP.
+See URL `https://psalm.dev/'."
+  :command ("psalm.phar " "--no-progress" "--output-format=emacs"
```
Comment (zonuexe) [inline flycheck.el:10095]: This command is not the best specification.
 
[Installation - Documentation](https://psalm.dev/docs/running_psalm/installation/) describes how to install Psalm on each project basis.  We can never assume that the file `psalm.phar` is in `PATH`.

File: flycheck.el [hunk 2]
```diff
@@ -10077,6 +10078,31 @@ See URL `https://github.com/uber/prototool'."
...
+            (option "--threads=" flycheck-psalm-threads)
+            (eval
+             (concat "--config=" (flycheck-psalm-project-root) "psalm.xml"))
+            (eval (buffer-file-name)))
```
Comment (zonuexe) [inline flycheck.el:10099]: Probably this line will be replaced by `source-inplace` or `source-original`.

File: flycheck.el (hunk 1 above)
Comment (zonuexe) [reply flycheck.el:10095]: `phpstan.el` takes a slightly complicated approach to solving that problem and Docker support.

https://github.com/emacs-php/phpstan.el/blob/master/phpstan.el#L262-L286

**Comment by zonuexe** [comment]
I created [new `psalm.el`](https://github.com/emacs-php/psalm.el) which is a fork of my [`phpstan.el`](https://github.com/emacs-php/phpstan.el).

Can you cooperate with us?

File: doc/languages.rst [hunk 3]
```diff
@@ -929,6 +929,14 @@ to view the docstring of the syntax checker.  Likewise, you may use
          The coding standard, either as name of a built-in standard, or as path
          to a standard specification.
 
+   .. syntax-checker:: psalm
+
+      Lint with `Psalm static analysis tool <https://psalm.dev/>`_.
+
+      .. defcustom:: flycheck-psalm-threads
```
Comment (zonuexe) [inline doc/languages.rst:936]: Maybe it doesn't make sense to specify the number of threads to analyze a single file.

**Comment by Sasanidas** [comment]
> I created [new `psalm.el`](https://github.com/emacs-php/psalm.el) which is a fork of my [`phpstan.el`](https://github.com/emacs-php/phpstan.el).
> 
> @Sasanidas Can you cooperate with us?

Sure, should I close this pull request then?

Thank you.

**Comment by zonuexe** [comment]
@Sasanidas 
I submitted `psalm` and `flycheck-psalm` recipe to MELPA. https://github.com/melpa/melpa/pull/6897

You can close this PR.  Thanks.

**Pull request closed without merging**